rfcs/0017-long-fork-consensus.md

```diff
@@ -0,0 +1,105 @@
+## Summary
+[summary]: #summary
+
+We currently do not support long forks. This means that to join the network you need a recent checkpoint, in other words to have always been online or to trust another party that has always been online.
+
+This adds support for a client to join the network at any point of time, regardless of if its been online recently. It adds this safely under a [50% + epsilon] security assumption. It also includes a mechanism to restore a chain to strength if this assumption was to be temporarily violated.
+
+This mechanism also does not require checkpointing at any time horizon, compared to alternatives.
+
+## Motivation
+
+[motivation]: #motivation
+
+It's not realistic for clients to have always been online, and it is a poor trust model to require clients to find a trusted third party (high friction and not always feasible).
+
+With this feature, clients will be able to join the network from any device at any time.
+
+## Detailed design
+
+[detailed-design]: #detailed-design
+
+### Part 1: min-window
+
+This version of the implementation is correct given that [50% + epsilon] of stake has always been participating and honest. 
+
+* Add a new field to `protocol_state` called `min_window`, initally set to `8k`
+* Add a new field to `protocol_state` called `current_window_length`, initially set to 0
+* `let window_diff = FLOOR(current_protocol_state.global_slot_number/(28)) - FLOOR(previous_protocol_state.global_slot_number/(8k))`
+* if `window_diff == 0`
+  * set `current_protocol_state.current_window_length` to `previous_protocol_state.current_window_length + 1`
+* else if `window_iff == 1`
+  * set `current_protocol_state.min_window` to `MIN(current_protocol_state.current_window_length, previous_protocol_state.min_window)`
+  * set `current_protocol_state.current_window_length` to `1`
+* else
+  * set `current_protocol_state.min_window` to `0`
+  * set `current_protocol_state.current_window_length` to `1`
+
+Add to the chain select(A,B) function:
+
+* If the previous-epoch checkpoint does not match between current and proposed, choose the chain with the greater `min_`. If its a tie choose the already held chain.
```

Comment:
s/min_/min_window

Reply:
it should use the existing tiebreaker logic in the event of a tie in `min_window`. using the already held chain will lead to inconsistency when different nodes get the blocks in different orders.